pkg/yurttunnel/pki/certmanager/certmanager.go

```diff
@@ -134,21 +153,30 @@ func newCertManager(
 	organizations,
 	dnsNames []string,
 	keyUsages []certificates.KeyUsage,
-	ipAddrs []net.IP) (certificate.Manager, error) {
+	ips []net.IP,
+	getIPs serveraddr.GetIPs) (certificate.Manager, error) {
 	certificateStore, err :=
 		store.NewFileStoreWrapper(componentName, certDir, certDir, "", "")
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize the server certificate store: %v", err)
 	}
 
 	getTemplate := func() *x509.CertificateRequest {
+		// use dynamic ips
+		if getIPs != nil {
+			tmpIPs, err := getIPs()
+			if err == nil && len(tmpIPs) != 0 && !reflect.DeepEqual(tmpIPs, ips) {
```

Comment:
we need to remove condition `!reflect.DeepEqual(tmpIPs, ips)` because `getTemplate` will verify the changes.